src/morphology.rs

```diff
@@ -427,13 +428,12 @@ impl Mask {
             image.height() < 512,
             "the input image must be at most 511 pixels high"
         );
-        Self {
-            elements: (0..image.width() as i16)
-                .cartesian_product(0..(image.height() as i16))
-                .filter(|(x, y)| image.get_pixel(*x as u32, *y as u32).0[0] != 0)
-                .map(|(x, y)| (x - center_x as i16, y - center_y as i16))
-                .collect(),
-        }
+        let elements = image
+            .enumerate_pixels()
+            .filter(|(_, _, &p)| p[0] != 0)
+            .map(|(x, y, _)| Point::new(x as i16 - center_x as i16, y as i16 - center_y as i16))
+            .collect();
+        Self { elements }
```

Comment:
Add private `new` and call it instead of `Self { elements }` everywhere

```
fn new(elements: Vec<Point<i16>>) -> Self {
    assert!(elements.len() <= (511 * 511) as usize);
    Self { elements } 
}
```

Reply:
done !